docs/pages/enroll-resources/kubernetes-access/troubleshooting.mdx

```diff
@@ -276,3 +276,138 @@ resource with the `create` verb, update it to allow the `get` verb as well:
 
 Once the `ClusterRole` is updated, you should be able to exec into Pods using
 `kubectl` version 1.30 or later.
+
+## Health Check: Unable to contact the Kubernetes cluster
+
+### Symptoms
+
+A Kubernetes cluster is reported as `unhealthy`, and all calls between Teleport and the Kubernetes cluster API fail.
+
+### Explanation
+
+A Teleport Kubernetes Service performed multiple health checks on a Kubernetes cluster and did not receive a response.
+
+Calls to Kubernetes cluster API endpoints `/selfsubjectaccessreviews` and `/readyz` returned errors without HTTP status codes.
+
+### Resolution
+
+Examine the Kubernetes cluster and network for root causes.
+
+Check the state of the Kubernetes cluster.
+```bash
+kubectl cluster-info
+kubectl get --raw /readyz
+```
+
+Check network and firewall rules.
+- Is the network configured properly?
+- Check security groups
+- Check network policies
+
+## Health Check: Missing required Kubernetes permissions
+
+### Symptoms
+
+A Kubernetes cluster is reported as `unhealthy`, and is missing one or more Kubernetes permissions such as `impersonate users`, `impersonate groups`, `impersonate service accounts` and/or `get pods`.
+
+### Explanation
+
+A Teleport Kubernetes Service performed a health check on a Kubernetes cluster, and received a response saying required Kubernetes cluster capabilities are denied.
+
+The Kubernetes cluster is missing a `ClusterRole`, or is configured with a partial `ClusterRole`. Teleport requires four Kubernetes [SelfSubjectAccessReview](https://kubernetes.io/docs/reference/kubernetes-api/authorization-resources/self-subject-access-review-v1/) permissions:
+- Impersonate users
+- Impersonate groups
+- Impersonate service accounts
+- Get pods
+
+### Resolution
+
+Add a `ClusterRole` and `ClusterRoleBinding` to the Kubernetes cluster with the four permissions.
+
+```bash
+kubectx <your-kube-context>
```

Comment:
TIL about `kubectx`!

However for a guide, we should stick to standard tooling IMO

```suggestion
$ kubectl config use-context <your-kube-context>
```